File: CodeWars/JavaScript/Rot13.js
// https://www.codewars.com/kata/530e15517bc88ac656000716/train/javascript
function rot13(message){
    const alphabet = "abcdefghijklmnopqrstuvwxyz"
    return message.split('').map((m) => {
        let capital = false
        if (alphabet.indexOf(m) === -1) {
            if (alphabet.includes(m.toLowerCase())) {capital = !capital}
            else {return m}
        }
        const l = m.toLowerCase()
        const cipher = alphabet.indexOf(l)+13 > 25? alphabet[alphabet.indexOf(l)-13] : alphabet[alphabet.indexOf(l)+13]
        return capital? cipher.toUpperCase() : cipher
    }).join('')
}

console.log(rot13("Grfg 10"))

File: CodeWars/Python/rot_13.py
## https://www.codewars.com/kata/530e15517bc88ac656000716/train/python
def rot13(message):
    A = ord('A')
    Z = ord('Z')
    a = ord('a')
    z = ord('z')
    m = list(message)
    
    def cipher(l):
        capital = False
        if ord(l) in range(A,Z+1):
            capital = True
        if ord(l.lower()) in range(a,z+1):
            sub = ord(l.lower())+13
            if sub > z:
                sub = a + (sub-z-1)
            if capital == True:
                return chr(sub).upper()
            else:
                return chr(sub)
        else:
            return l
                
    return ''.join(list(map(cipher, m)))

rot13("Grfg")